pallet/doughnut/src/lib.rs

```diff
@@ -313,6 +318,8 @@ pub mod pallet {
 		TRNNutPermissionDenied,
 		/// Inner call is not whitelisted
 		UnsupportedInnerCall,
+		/// Not a whitelisted holder
+		NotAWhitelistedHolder,
```

Comment:
nit
```suggestion
		HolderNotWhitelisted,
```
or "HolderNotAllowed"